UnitsNet/CustomCode/UnitParser.cs

```diff
@@ -54,13 +54,14 @@ internal static class UnitParser
 
             const string exponentialRegex = @"(?:[eE][-+]?\d+)?)";
 
-            string regexString = string.Format(@"(?:\s*(?<value>[-+]?{0}{1}{2}{3})?{4}{5}",
+            string regexString = string.Format(@"(?:\s*(?<value>[-+]?{0}{1}{2}{3})?",
                 numRegex, // capture base (integral) Quantity value
                 exponentialRegex, // capture exponential (if any), end of Quantity capturing
                 @"\s?", // ignore whitespace (allows both "1kg", "1 kg")
-                @"(?<unit>[^\s\d,]+)", // capture Unit (non-whitespace) input
-                @"(and)?,?", // allow "and" & "," separators between quantities
-                @"(?<invalid>[a-z]*)?"); // capture invalid input
+                @"(?<unit>[^\d]+)"); // capture Unit (non-numeric)
```

Comment:
Ah, you raise some good points here.

> I think you're mistaking TUnit in UnitParser.ParseUnit() as the Unit Enums

You are right, I was probably too quick here, but `UnitSystem` should be able to provide this information as you found out.

>  It will fail on test "1ft 1invalid" since it recognizes the "INvalid" as inches. It also fails the test "1ft monkey 1in" since it recognizes "FT monkey". It also fails the test "1ft monkey 1in" since it recognizes "FT monkey".
I thought about restricting it to match only exact abbreviation but how will it handle something like "m" and "mm"?

I think your suggestion should work, we just need ensure we don't match on text that is not part of a valid pair of `{value}{unit}` or the accepted delimiters between pairs or between value and unit.  I tried a few things and wound up with this: `^((?<value>\d+)\s?(?<unit>m²|m\^2|cm²|cm\^2|ft|in|m|mm)\s?)+$` and I added unit tests for the cases you listed above that you can view here: https://regex101.com/r/EMLIwt/1

With the start/end boundaries, a `+` repeater to match on at least one pair and an optional whitespace `\s?` between each pair, the regex now seems to properly match on full abbreviations, and it should not incorrectly match `m` on `1mm`.

> I think having a single regex or parse algorithm for all the Unit types will have its limitations.
If we want the parsing to handle special cases then maybe it will be easier to implement one parse method for each Unit type.

I hear you and I agree that might be something we eventually have to reach for, but I also do think my updated regex might just work. The upside of the complexity is that it allows for parsing any number of valid pairs of value+unit and that the implementation is uniform across units, so it is consistent and easier to reason about. The downside is the complexity of the regex itself, performance and the fact that it might not make sense or at least not be commonly used to parse all kinds of pairs of quantities from a string, like we are used to from `1ft 2in` and `1' 2''`.

What do you think?